zgrab_schema.py

```diff
@@ -205,7 +205,64 @@
         "value":Binary(),
         "length":Integer(),
         "lifetime_hint":Long()
-    })
+    }),
+    "key_material":SubRecord({
+        "pre_master_secret":SubRecord({
+            "value":Binary(),
+            "length":Integer()
+        }),
+        "master_secret":SubRecord({
+            "value":Binary(),
+            "length":Integer()
+        }),
+    }),
+    "client_finished":SubRecord({
+        "verify_data":Binary()
+    }),
+    "client_key_exchange":SubRecord({
+        "dh_params":SubRecord({
+            "prime":SubRecord({
+                "value":Binary(),
+                "length":Integer()
+            }),
+            "generator":SubRecord({
+                "value":Binary(),
+                "length":Integer()
+            }),
+            "client_public":SubRecord({
+                "value":Binary(),
+                "length":Integer()
+            }),
+            "client_private":SubRecord({
+                "value":Binary(),
+                "length":Integer()
+            }),
+        }),
+        "ecdh_params":SubRecord({
+            "curve_id":SubRecord({
+                "name":String(),
+                "id":Integer()
+            }),
+            "client_public":SubRecord({
+                "x":SubRecord({
+                    "value":Binary(),
+                    "length":Integer()
+                }),
+                "y":SubRecord({
+                    "value":Binary(),
+                    "length":Integer()
+                }),
+            }),
+            "client_private":SubRecord({
+                "value":Binary(),
+                "length":Integer()
+            }),
+        }),
+        "rsa_params":SubRecord({
+            "length":Integer(),
+            "encrypted_pre_master_secret":Binary()
```

Comment:
No, this is the encrypted version sent over the wire to the server. The unencrypted premaster secret is included in the key_material record and removed here IOT reduce duplication. With the --tls-verbose flag, you can't output the client KEX record w/o the key material record.

```golang
func (m *clientKeyExchangeMsg) MakeLog(ka keyAgreement) *ClientKeyExchange {
 	ckx := new(ClientKeyExchange)
 	ckx.Raw = make([]byte, len(m.raw))
 	copy(ckx.Raw, m.raw)
 
 	switch ka := ka.(type) {
 	case *rsaKeyAgreement:
 		ckx.RSAParams = new(keys.RSAClientParams)
 		ckx.RSAParams.Length = uint16(len(m.ciphertext) - 2) // First 2 bytes are length
 		ckx.RSAParams.EncryptedPMS = make([]byte, len(m.ciphertext)-2)
 		copy(ckx.RSAParams.EncryptedPMS, m.ciphertext[2:])
 		// Premaster-Secret is available in KeyMaterial record
 	case *dheKeyAgreement:
 		ckx.DHParams = ka.ClientDHParams()
 	case *ecdheKeyAgreement:
 		ckx.ECDHParams = ka.ClientECDHParams()
 	default:
 		break
 	}
 
 	return ckx
 }
```